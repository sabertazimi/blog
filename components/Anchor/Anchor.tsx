import type { AnchorProps } from 'antd';
import { Anchor as AntAnchor } from 'antd';

interface Props extends AnchorProps {}

const Anchor = (props: Props): JSX.Element => <AntAnchor {...props} />;

export default Anchor;
