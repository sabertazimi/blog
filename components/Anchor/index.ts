export { Anchor, Link } from './Anchor';
