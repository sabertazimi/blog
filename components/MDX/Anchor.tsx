import { Typography } from 'antd';

interface Props {}

const Anchor = (props: Props): JSX.Element => <Typography.Link {...props} />;

export default Anchor;
