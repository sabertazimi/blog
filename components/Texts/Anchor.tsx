import { cx } from '@components/utils'
import { Typography } from 'antd'
import type { ReactNode } from 'react'
import styles from './Anchor.module.css'

interface Props {
  children?: ReactNode
  className?: string
}

function Anchor({ className, ...props }: Props): JSX.Element {
  return (
    <Typography.Link
      {...props}
      className={cx('mdx-hash-link', className, styles.anchor)}
    />
  )
}

export default Anchor
