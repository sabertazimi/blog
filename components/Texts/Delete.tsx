import Text from './Text'

interface Props {}

const Delete = (props: Props): JSX.Element => <Text {...props} delete />

export default Delete
