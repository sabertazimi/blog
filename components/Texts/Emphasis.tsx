import Text from './Text'

interface Props {}

const Emphasis = (props: Props): JSX.Element => <Text {...props} italic />

export default Emphasis
