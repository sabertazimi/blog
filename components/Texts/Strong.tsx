import Text from './Text'

interface Props {}

const Strong = (props: Props): JSX.Element => <Text {...props} strong />

export default Strong
