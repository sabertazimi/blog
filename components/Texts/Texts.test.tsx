import { render } from '@utils';
import Anchor from './Anchor';
import Delete from './Delete';
import Emphasis from './Emphasis';
import Strong from './Strong';

describe('Texts', () => {
  const Texts = {
    a: Anchor,
    strong: Strong,
    em: Emphasis,
    del: Delete,
  };

  test.each(Object.values(Texts))(
    'should render correctly (snapshot)',
    Text => {
      const { container } = render(<Text>Text</Text>);

      expect(container).toMatchSnapshot();
    }
  );
});
