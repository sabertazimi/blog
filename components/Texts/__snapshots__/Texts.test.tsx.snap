// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Texts should render correctly (snapshot) 1`] = `
<div>
  <a
    class="ant-typography mdx-hash-link anchor"
  >
    Text
  </a>
</div>
`;

exports[`Texts should render correctly (snapshot) 2`] = `
<div>
  <span
    class="ant-typography dark:text-light"
  >
    <strong>
      Text
    </strong>
  </span>
</div>
`;

exports[`Texts should render correctly (snapshot) 3`] = `
<div>
  <span
    class="ant-typography dark:text-light"
  >
    <i>
      Text
    </i>
  </span>
</div>
`;

exports[`Texts should render correctly (snapshot) 4`] = `
<div>
  <span
    class="ant-typography dark:text-light"
  >
    <del>
      Text
    </del>
  </span>
</div>
`;
